homeassistant/components/lyric/climate.py

```diff
@@ -0,0 +1,305 @@
+        """Initialize Honeywell Lyric climate entity."""
+        self._temperature_unit = hass.config.units.temperature_unit
+
+        # Setup supported hvac modes
+        self._hvac_modes = [HVAC_MODE_OFF]
+
+        # Add supported lyric thermostat features
+        if LYRIC_HVAC_MODE_HEAT in device.allowedModes:
+            self._hvac_modes.append(HVAC_MODE_HEAT)
+
+        if LYRIC_HVAC_MODE_COOL in device.allowedModes:
+            self._hvac_modes.append(HVAC_MODE_COOL)
+
+        if (
+            LYRIC_HVAC_MODE_HEAT in device.allowedModes
+            and LYRIC_HVAC_MODE_COOL in device.allowedModes
+        ):
+            self._hvac_modes.append(HVAC_MODE_HEAT_COOL)
+
+        super().__init__(
+            coordinator,
+            location,
+            device,
+            f"{device.macID}_thermostat",
+            device.name,
+            "mdi:thermostat",
+        )
+
+    @property
+    def supported_features(self) -> int:
+        """Return the list of supported features."""
+        return SUPPORT_FLAGS
+
+    @property
+    def temperature_unit(self) -> str:
+        """Return the unit of measurement."""
+        return self._temperature_unit
+
+    @property
+    def current_temperature(self) -> Optional[float]:
+        """Return the current temperature."""
+        device: LyricDevice = self.coordinator.data.locations_dict[
+            self._location.locationID
+        ].devices_dict[self._device.macID]
+        return device.indoorTemperature
+
+    @property
+    def hvac_mode(self) -> str:
+        """Return the hvac mode."""
+        device: LyricDevice = self.coordinator.data.locations_dict[
+            self._location.locationID
+        ].devices_dict[self._device.macID]
+        return HVAC_MODES[device.changeableValues.mode]
+
+    @property
+    def hvac_modes(self) -> List[str]:
+        """List of available hvac modes."""
+        return self._hvac_modes
+
+    @property
+    def target_temperature(self) -> Optional[float]:
+        """Return the temperature we try to reach."""
+        device: LyricDevice = self.coordinator.data.locations_dict[
+            self._location.locationID
+        ].devices_dict[self._device.macID]
+        if not device.hasDualSetpointStatus:
+            return device.changeableValues.heatSetpoint
+
+    @property
+    def target_temperature_low(self) -> Optional[float]:
+        """Return the upper bound temperature we try to reach."""
+        device: LyricDevice = self.coordinator.data.locations_dict[
+            self._location.locationID
+        ].devices_dict[self._device.macID]
+        if device.hasDualSetpointStatus:
+            return device.changeableValues.coolSetpoint
+
+    @property
+    def target_temperature_high(self) -> Optional[float]:
+        """Return the upper bound temperature we try to reach."""
+        device: LyricDevice = self.coordinator.data.locations_dict[
+            self._location.locationID
+        ].devices_dict[self._device.macID]
```

Comment:
Make this a property so you can do `self.device.hasDualSetpointStatus`

Reply:
Of course! That simplifies things, thanks :smiley_cat: